src/SprykerSdk/Integrator/Executor/ManifestExecutor.php

```diff
@@ -0,0 +1,163 @@
+
+namespace SprykerSdk\Integrator\Executor;
+
+use RuntimeException;
+use SprykerSdk\Integrator\Dependency\Console\InputOutputInterface;
+use SprykerSdk\Integrator\Manifest\ManifestReader;
+use SprykerSdk\Integrator\ManifestStrategy\ManifestStrategyInterface;
+use SprykerSdk\Integrator\SprykerLock\SprykerLockReader;
+use SprykerSdk\Integrator\SprykerLock\SprykerLockWriter;
+
+class ManifestExecutor
+{
+    /**
+     * @var \SprykerSdk\Integrator\SprykerLock\SprykerLockReader
+     */
+    protected $sprykerLockReader;
+
+    /**
+     * @var \SprykerSdk\Integrator\Manifest\ManifestReader
+     */
+    protected $manifestReader;
+
+    /**
+     * @var \SprykerSdk\Integrator\ManifestStrategy\ManifestStrategyInterface[]
+     */
+    protected $manifestExecutors;
+
+    /**
+     * @var \SprykerSdk\Integrator\SprykerLock\SprykerLockWriter
+     */
+    protected $sprykerLockWriter;
+
+    /**
+     * @param \SprykerSdk\Integrator\SprykerLock\SprykerLockReader $sprykerLockReader
+     * @param \SprykerSdk\Integrator\Manifest\ManifestReader $manifestReader
+     * @param \SprykerSdk\Integrator\SprykerLock\SprykerLockWriter $sprykerLockWriter
+     * @param \SprykerSdk\Integrator\ManifestStrategy\ManifestStrategyInterface[] $manifestExecutors
+     */
+    public function __construct(
+        SprykerLockReader $sprykerLockReader,
+        ManifestReader $manifestReader,
+        SprykerLockWriter $sprykerLockWriter,
+        array $manifestExecutors
+    ) {
+        $this->sprykerLockReader = $sprykerLockReader;
+        $this->manifestReader = $manifestReader;
+        $this->manifestExecutors = $manifestExecutors;
+        $this->sprykerLockWriter = $sprykerLockWriter;
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ModuleTransfer[] $moduleTransfers
+     * @param \SprykerSdk\Integrator\Dependency\Console\InputOutputInterface $inputOutput
+     * @param bool $isDry
+     *
+     * @return int
+     */
+    public function runModuleManifestExecution(array $moduleTransfers, InputOutputInterface $inputOutput, bool $isDry): int
+    {
+        $this->assertModuleData($moduleTransfers);
+
+        $manifests = $this->manifestReader->readManifests($moduleTransfers);
+
+        $sprykerLock = $this->sprykerLockReader->getLockFileData();
+
+        $unappliedManifests = $this->findUnappliedManifests($manifests, $sprykerLock);
+
+        if (!$unappliedManifests) {
+            return 0;
+        }
+
+        if (!$inputOutput->confirm('There are unapplied manifests found for your modules. Do you want to apply them?')) {
+            return 0;
+        }
+
+        $GLOBALS['IO'] = $inputOutput;
+
+        foreach ($unappliedManifests as $moduleName => $moduleManifests) {
+            foreach ($moduleManifests as $manifestType => $unappliedManifestByType) {
+                $manifestExecutor = $this->resolveExecutor($manifestType);
+
+                foreach ($unappliedManifestByType as $manifestHash => $unappliedManifest) {
```

Comment:
would be nice to move smth to separate methods in order to decrease code complexity

Reply:
still valid

Reply:
and still valid